Automattic/jetpack
# Fix: Display error for broken social connections in editor

#### Changes proposed in this Pull Request:


This change introduces a notice if there is any broken connection, and disables broken connections 
Added new property to connection meta to state health of connection. We already know this information, since this is cached for 12 hours, so we can easily use it in the requests. This also had an advantage, that in the editor the Notice is instantly visible, rather than before when it sent a request on page load to get the (same cached) connection results from the API, and then showing the notice.

#### Jetpack product discussion



1202807135507995-as-1202841973973991

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Create a broken connection for facebook: 
  * Go [here](https://www.facebook.com/settings?tab=business_tools&ref=business_login_reentry) and revoke access for WordPress
  * In sandbox open wp-content/lib/external-connections.php file, and on line 655 change `$force = false` to `$force = true`
  * Open Social on your page, then go to Add new Post, and open the Social/Jetpack tab. This should refresh the test result and after that in cloud/Calypso the FB connection should show as broken.
  * Revert the change in the parameters, set it back to false. Now you have a broken connection without have to wait for 12 hours.
* Open Gutenberg, and go to Social tab. Facebook connection should be shown as broken and disabled:


Gutenberg            |  Classic
:-------------------------:|:-------------------------:
<img width="279" alt="CleanShot 2022-08-24 at 14 00 45@2x" src="https://user-images.githubusercontent.com/36671565/186413178-4b9b55ab-8015-49e7-82f4-257efe8e2c31.png"> |  <img width="290" alt="CleanShot 2022-08-29 at 14 27 16@2x" src="https://user-images.githubusercontent.com/36671565/187200853-3848aab9-d97a-4225-8f62-abf7eeb825dd.png">>


* Click on link, which navigates to JP cloud, and fix the connection.
* Going back to the Gutenberg tab, the connection should refresh in a few seconds. Notice should disappear, and connection can be toggled again. In Classic Editor you need to refresh the page to see changes
* Check this for both Social standalone and Jetpack version.
